### PR TITLE
Add CNI / CPI / CSI / CoreDNS helm releases to cluster-azure

### PR DESCRIPTION
**Pull request by primeroz**
Towards https://github.com/giantswarm/roadmap/issues/2271

Requires https://github.com/giantswarm/default-apps-azure/pull/78

### Please check if PR meets these requirements

- [ ] Results of the diffs have been examined and no unintended changes are being introduced.

### Helper

* to disable the GH Action generating manifests diffs for installations, between target and source branches, comment the `/no_diffs_printing` on the PR.

### Trigger e2e tests

Cannot run CI because it needs a branch of default-apps-azure

### CODE REVIEW

File: helm/cluster-azure/templates/helmreleases/cleanup-helmreleases-hook-job.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        runAsUser: 1000
+        runAsGroup: 1000
+      containers:
+        - name: post-delete-job
+          image: "{{ .Values.internal.kubectlImage.registry }}/{{ .Values.internal.kubectlImage.name }}:{{ .Values.internal.kubectlImage.tag }}"
+          command:
+            - "/bin/sh"
+            - "-xc"
+            - |
+              for r in $(kubectl get helmrelease -n {{ .Release.Namespace }} -l "giantswarm.io/cluster={{ include "resource.default.name" . }}" -o name) ; do
```
Comment (fiunchinho) [inline helm/cluster-azure/templates/helmreleases/cleanup-helmreleases-hook-job.yaml:111]: [We recently updated the job in CAPA to also clean `HelmCharts`](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/cleanup-helmreleases-hook-job.yaml#L117). This needs extra RBAC permissions btw.

File: helm/cluster-azure/templates/helmreleases/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{/*
+Common Values for helm releases
+*/}}
+{{- define "values.common" -}}
+provider: capz
+{{- end -}}
+
```
Comment (fiunchinho) [inline helm/cluster-azure/templates/helmreleases/_helpers.tpl:9]: Is this really necessary? It's only used once, so I'd inline `provider: capz` where is needed.

Comment (primeroz) [reply helm/cluster-azure/templates/helmreleases/_helpers.tpl:9]: I thought there might be more going forward but sure :+1:

File: helm/cluster-azure/templates/helmreleases/cleanup-helmreleases-hook-job.yaml (hunk 1 above)
Comment (primeroz) [reply helm/cluster-azure/templates/helmreleases/cleanup-helmreleases-hook-job.yaml:117]: done

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**